src/NServiceBus.Core/Licensing/LicenseManager.cs

```diff
@@ -59,6 +70,65 @@ internal void InitializeLicense(string licenseText)
             license = foundLicense;
         }
 
+        static bool FirstTimeUser()
+        {
+            // Check for the presence of HKCU:Software/NServiceBus
+            using (var regNsbRoot = Registry.CurrentUser.OpenSubKey(@"Software\NServiceBus"))
+            {
+                if (regNsbRoot == null)
+                {
+                    //Check for the presence of HKCU:SOFTWARE\ParticularSoftware
+                    using (var regParticularRoot = Registry.CurrentUser.OpenSubKey(@"Software\ParticularSoftware"))
+                    {
+                        if (regParticularRoot == null)
+                        {
+                            return true;
+                        }
+                    }
+                }
+            }
+            return false;
+        }
+
+
+        private void TrackFirstTimeUsageEvent()
+        {
+            // Get the current version of NServiceBus that's being used
+            var assembly = Assembly.GetExecutingAssembly();
+            var version = assembly.GetName().Version.ToString();
+
+            if (assembly.Location != null)
+            {
+                version = FileVersionInfo.GetVersionInfo(assembly.Location).FileVersion;
+            }
+
+            try
+            {
+                var azureFuncUrl =
+                    "https://first-time-nuget-user.azurewebsites.net/api/TrackNewUser?code=1bkjzkbyeuf1ed46l87yom9529yu5ykzt81mjx65kgtmmv4pldiyxzvfprxf9s3xqxykf279cnmi";
+
+                var postData = $"{{\"version\":\"{version}\"}}";
+                var client = new HttpClient();
+                client.DefaultRequestHeaders.Accept.Add(new MediaTypeWithQualityHeaderValue("application/json"));
+
+                var request = new HttpRequestMessage(HttpMethod.Post, azureFuncUrl);
+                request.Content = new StringContent(postData, Encoding.UTF8, "application/json");
+                var response = client.SendAsync(request).GetAwaiter().GetResult();
```

Comment:
@SimonCropp @danielmarbach - Originally I was calling this method inside a dialog. But that is no longer an issue. I am going to change the code to something like the below, where the code won't block and will log on exception. 

So essentially, going to change the code to:

client.SendAsync(request).ContinueWith(t => Logger.WarnFormat("Could not report first time usage stat to www.particular.net"),
        TaskContinuationOptions.OnlyOnFaulted);